src/Runtime/OMInstrumentHelper.h

```diff
@@ -0,0 +1,108 @@
+
+//===----------------------------------------------------------------------===//
+// Timing support for MVS
+
+#ifdef __MVS__
+#define timersub(a, b, result)                                                 \
+  do {                                                                         \
+    (result)->tv_sec = (a)->tv_sec - (b)->tv_sec;                              \
+    (result)->tv_usec = (a)->tv_usec - (b)->tv_usec;                           \
+    if ((result)->tv_usec < 0) {                                               \
+      --(result)->tv_sec;                                                      \
+      (result)->tv_usec += 1000000;                                            \
+    }                                                                          \
+  } while (0);
+#endif
+
+//===----------------------------------------------------------------------===//
+// Timing functions
+
+#if OM_DRIVER_TIMING && !OM_DRIVER_TIMING_DISABLE_ALL
+#include <stdio.h>
+#include <sys/time.h>
+
+// Global variable to help OMInstrumentHelper.h to keep track of nesting level
+// of timing operations.
+extern int timing_nest_level;
+extern char timing_nest_strings[6][20];
+
+#define TIMING_INIT(_var_name)                                                 \
+  /* Define variable in current scope. */                                      \
+  struct timeval _var_name, _var_name##_tmp;                                   \
+  int _var_name##_nest_level = -1;                                             \
+  _var_name.tv_sec = 0;                                                        \
+  _var_name.tv_usec = 0;
+
+#define TIMING_START(_var_name)                                                \
+  _var_name##_nest_level = timing_nest_level;                                  \
+  ++timing_nest_level;                                                         \
+  gettimeofday(&_var_name##_tmp, NULL);
+
+#define TIMING_STOP(_var_name)                                                 \
+  { /* Define variables in their own scope */                                  \
+    struct timeval start_time, stop_time, diff_time;                           \
+    start_time = _var_name##_tmp;                                              \
+    gettimeofday(&stop_time, NULL);                                            \
+    timersub(&stop_time, &start_time, &diff_time);                             \
+    _var_name.tv_sec += diff_time.tv_sec;                                      \
+    _var_name.tv_usec += diff_time.tv_usec;                                    \
+    --timing_nest_level;                                                       \
+  }
+
+#define TIMING_PRINT(_var_name)                                                \
+  if (_var_name##_nest_level >= 0) { /* was started at least once */           \
+    int l = _var_name##_nest_level <= 5 ? _var_name##_nest_level : 5;          \
+    fprintf(stderr, "@OM_DRIVER, %s%s, %ld.%06ld\n", timing_nest_strings[l],   \
```

Comment:
We can avoid using `timing_nest_string` by using `%*` to print out "" `l` times, e.g.
```c
fprintf(stderr, "@OM_DRIVER, %*s%s, %ld.%06ld\n", l, "",
           #_var_name, (long int)_var_name.tv_sec, (long int)_var_name.tv_usec);
```

Reply:
I will give this pattern a try.